types/V1.mli

```diff
@@ -125,40 +125,38 @@ module type FLOW = sig
       logging. *)
 
   val read: flow -> [`Ok of buffer | `Eof | `Error of error ] io
-  (** [read flow] will block until it either successfully reads a
-      segment of data from the current flow, receives an [Eof]
-      signifying that the connection is now closed, or an [Error]. *)
+  (** [read flow] reads some data from the flow and returns a
+      fresh buffer (of arbitrary size), an [`Eof] if the other side has
+      called [close] or an [`Error]. *)
```

Comment:
in the previous comment I read `block` and assume blocking behaviour if no data is there to be read; whereas the new does not specify the blocking semantics.  since basically all `io` functions are blocking, it might not be worth to mention it at every function, but in some general introduction at the top!?